limacharlie/Extensions.py

```diff
@@ -57,7 +42,72 @@ def delete( self, extName ):
+    
+    def migrate( self, extName ):
+        return self._manager._apiCall( 'extension/migrate/%s' % ( extName, ), POST, {
+            'oid' : self._manager._oid,
+        } )
 
+    def request( self, extName, action, data = {}, isImpersonated = False ):
+        req = {
+            'oid' : self._manager._oid,
+            'action' : action,
+            'data' : json.dumps( data ),
+        }
+        if isImpersonated:
+            if self._manager._jwt is None:
+                self._manager._refreshJWT()
+            req[ 'impersonator_jwt' ] = self._manager._jwt
+        return self._manager._apiCall( 'extension/request/%s' % ( extName, ), POST, req )
+    
+    def convert_rules(self, extName, isDryRun = True): 
+        extList = self.list()
+        if extName not in extList:
+            return f'ERROR: unable to convert rules. {self._manager._oid} is not subscribed to {extName}'
+        gen_hive = Hive.Hive(self._manager, "dr-general")
+        man_hive = Hive.Hive(self._manager, "dr-managed")
+        gen_dr_rules = gen_hive.list()
+        man_dr_rules = man_hive.list()
+        merged_dr_rules = {**gen_dr_rules, **man_dr_rules}
+        updated_rules = []
+        for rule_name in merged_dr_rules:
+            is_general = rule_name in gen_dr_rules 
+            is_managed = rule_name in man_dr_rules
+            gen_hive_records = None
+            man_hive_records = None   
+            if is_general:
+                gen_hive_records = gen_hive.get(rule_name)
+            if is_managed:
+                man_hive_records = man_hive.get(rule_name)                   
+            dnr = getattr(gen_hive_records, 'data', None) if is_general else getattr(man_hive_records, 'data', None)
+            if dnr is not None:
+                hive_type = 'dr-general' if is_general else 'dr-managed'
+                detect = dnr.get('detect', None)
+                resp_items = dnr.get('respond', None)  
+                if resp_items is not None and detect is not None:
+                    if contains_action_name(resp_items, extName):  
+                        rule_data = update_rule(rule_name, dnr, detect, resp_items, hive_type, extName)
+                        updated_rules.append(rule_data)          
+        if isDryRun and len(updated_rules) > 0:
+            for updated_rule in updated_rules:
+                print(f"Dry run of change on rule '{updated_rule['r_name']}':")
+                print("\033[91m- {}\033[0m".format(updated_rule['old_dnr'])) # print red text
+                print("\033[92m+ {}\033[0m".format(updated_rule['new_dnr'])) # print green text
+        if not isDryRun and len(updated_rules) > 0:
+            for updated_rule in updated_rules:
+                data = {
+                    "data": updated_rule['new_dnr']
+                }
+                try:
+                    hr = Hive.HiveRecord(updated_rule['r_name'], data)
```

Comment:
Same here Hive.HiveRecord does not have property HiveRecord 
You can do a import similar to that of Hive to bring HiveRecord into scope 
from limacharlie import HiveRecord


Then HiveRecord('','')